simpleviewinc/keg-cli
# Ip helper

## Context

* I have been wanting a task for helping with ip addresses

## Updates
* `repos/cli-utils/src/network/getPrivateIPs.js`
  * gets all private ips for the `en0` interface
* `repos/cli-utils/src/network/getAddresses.js`
  * helper for filtering the results of `os.networkInterfaces()`
* `repos/cli-utils/src/network/getHostIP.js`
  * just renamed and moved, used to be `getInternalIp`
  * also updated it to use `mapFind` 
* `repos/cli-utils/src/network/getPublicIP.js`
  * gets the current machine's public ip
* `repos/cli-utils/src/network/getPublicIPsForUrl.js`
  * returns the public ip for a given url
* `repos/cli-utils/src/network/ipIsInRange.js`
  * helper checking if an ip address falls within a range
* `src/tasks/network/ip.js`
  * new ip task for getting data about ip addresses

## Testing

* verify unit tests pass
* `keg net ip`
  * verify it prints out your public and one private ip
* `keg net ip --private`
   * verify it prints out only the private ip
* `keg net ip --public`
   * verify it prints out only the public ip
* `keg net ip --url archive.org`
  * verify it prints out the public ip for archive.or
